### ISSUE TITLE
[BUG]: Big Endian Guid problem

### ISSUE THREAD

**Issue by anatoliy-savchak** [aloneguid/parquet-dotnet#496]
### Library Version

4.23.4

### OS

Windows

### OS Architecture

64 bit

### How to reproduce?

1. Have parquet file, which contains Guid fields generated by other means like SQL Server CETAS. These are Big Endian guids.
2. Deserialize. Guid values are incorrect.
3. The fix could be to add new option => UseBigEndianForGuids

### Failing test

```csharp
C# Script in Polyglot Notebooks. But it works same in C# Console app:
#r "nuget:Parquet.Net" // remove for C# Console app

var msg = await Task.Run(() => (new System.Net.Http.HttpClient()).GetAsync("https://github.com/anatoliy-savchak/parquet-dotnet-guids/raw/main/cetas4.parquet"));
Parquet.Rows.Table tab = await Parquet.ParquetReader.ReadTableFromStreamAsync(msg.Content.ReadAsStream());
Console.WriteLine("{'Id': 15A2501E-4899-4FF8-AF51-A1805FE0718F} - Expected, but found:");
Console.WriteLine(tab.ToString());
```

**Comment by anatoliy-savchak** [comment]
Adding CETAS for anybody interested:
```
drop table if exists #mytable;
create table #mytable(Id uniqueidentifier);
insert into #mytable(Id) values ('15A2501E-4899-4FF8-AF51-A1805FE0718F');
--drop external exists cetas4;
create external table cetas4
with (
	location = 'cetas4/',
	data_source = DataSourceDatalake,
	file_format = ParquetUncompressed
)
as
select top 1 * from #mytable;
```

**Comment by aloneguid** [comment]
can you provide an example of this parquet file

**Comment by anatoliy-savchak** [comment]
> can you provide an example of this parquet file

Yes, sure: https://github.com/anatoliy-savchak/parquet-dotnet-guids/raw/main/cetas4.parquet

**Comment by aloneguid** [comment]
[cetas4.zip](https://github.com/aloneguid/parquet-dotnet/files/14865978/cetas4.zip)
thanks, attaching here for convenience

**Comment by anatoliy-savchak** [comment]
My current workaround:
```
    public async Task WriteDataToParquetFile<T>(string relativePath, string fileName, IEnumerable<T> data)
    {
        if (BitConverter.IsLittleEndian)
        {
            var guidProperties = typeof(T).GetProperties().Where(p => p.PropertyType == typeof(Guid) || p.PropertyType == typeof(Guid?));
            foreach (var prop in guidProperties)
            {
                foreach (var item in data)
                {
                    var value = (Guid?)prop.GetValue(item);
                    if (value != null)
                    {
                        var guidBytes = value.Value.ToByteArray();
                        Array.Reverse(guidBytes, 0, 4);
                        Array.Reverse(guidBytes, 4, 2);
                        Array.Reverse(guidBytes, 6, 2);
                        var newValue = new Guid(guidBytes);
                        prop.SetValue(item, newValue);
                    }
                }
            }
        }

        // Serialize data to parquet file
        using var memoryStream = new MemoryStream();
        await ParquetSerializer.SerializeAsync(data, memoryStream);
        memoryStream.Position = 0;

```

In addition .Net 8 has Guid constructor which has bigEndian arg.

![image](https://github.com/aloneguid/parquet-dotnet/assets/40269424/58b04a6b-6349-462c-9302-a8a782575065)

**Comment by anatoliy-savchak** [comment]
Good point, thanks @aloneguid !

We typically save data from C# BE into parquet in Azure Data Storage, and then read from SQL via external datasource or openrowset. SQL Server / Azure SQL  DB/Azure  SQL Server Managed Instance / SQL Server DW / SQL serverless pool (Synapse) etc all use Poly Base engine inside, which loads / saves using big-endian.

We are moving storage into ADS and converting all DTO's Guids into string is somewhat tedious. Plus additional overhead of converting varchar into uniqueidentifier on SQL side.

Currently we are going to use workaround above, but I hope the fix with Options is easy enough so we could continue using your awesome library! :)

**Comment by aloneguid** [comment]
thanks. Actually [specification](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#uuid) explicitly says it should be encoded as big-endian, so it's a bug here.

**Comment by aloneguid** [comment]
@anatoliy-savchak does this look correct?

![image](https://github.com/aloneguid/parquet-dotnet/assets/3155189/270694c8-d015-4bc1-b718-93d26f82ef6c)

**Comment by aloneguid** [comment]
The fix just released in 4.23.5, hopefully it works for you.
